### PR TITLE
Add details information to locations endpoint

### PR DESCRIPTION
**Pull request by yabirgb**
It was needed by the frontend to remove the mappings for locations in the frontend and that meant also to remove the hardocded images that they had.

### CODE REVIEW

File: rotkehlchen/types.py [hunk 1]
```diff
@@ -597,6 +598,24 @@ def to_chain_id(self) -> int:
 EVM_LOCATIONS: tuple[EVM_LOCATIONS_TYPE_, ...] = typing.get_args(EVM_LOCATIONS_TYPE_)
 
 
+class LocationDetails(NamedTuple):
```
Comment (LefterisJP) [inline rotkehlchen/types.py:601]: Add a docstring here explaining what this structure is

File: rotkehlchen/types.py [hunk 2]
```diff
@@ -597,6 +598,24 @@ def to_chain_id(self) -> int:
...
+    display_name: Optional[str] = None
+    icon: Optional[str] = None
+    image: Optional[str] = None
+
+    def serialize(self) -> dict[str, str]:
```
Comment (LefterisJP) [inline rotkehlchen/types.py:606]: you are not using asdict here only to have optional stuff not appear?

File: rotkehlchen/types.py [hunk 3]
```diff
@@ -597,6 +598,24 @@ def to_chain_id(self) -> int:
...
+        data = {}
+        if self.display_name is not None:
+            data = {'display_name': self.display_name}
+
+        if self.icon is not None:
+            return data | {'icon': self.icon}
+        if self.image is not None:
+            return data | {'image': self.image}
+
+        raise InputError('Location details does not have either icon or image')
```
Comment (LefterisJP) [inline rotkehlchen/types.py:616]:
```suggestion
        raise InputError('Location details has neither an icon nor an image')
```

File: rotkehlchen/detail_mappings.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
+from rotkehlchen.types import Location, LocationDetails
```
Comment (LefterisJP) [inline rotkehlchen/detail_mappings.py:1]: This should not be in the root of the project. Put it under `/constants`. Also name can be more specific. location_details.py

File: rotkehlchen/types.py (hunk 2 above)
Comment (yabirgb) [reply rotkehlchen/types.py:606]: Yes, the logic using asdict would be similar but using pop with if checks for None

**Review by LefterisJP — APPROVED** [review]
LGTM

**Pull request merged**